packages/@aws-cdk/aws-lambda/lib/log-retention.ts

```diff
@@ -39,16 +39,14 @@ export class LogRetention extends cdk.Construct {
       lambdaPurpose: 'LogRetention',
     });
 
-    if (provider.role && !provider.role.node.tryFindChild('DefaultPolicy')) { // Avoid duplicate statements
-      provider.role.addToPolicy(
-        new iam.PolicyStatement()
-          .addActions('logs:PutRetentionPolicy', 'logs:DeleteRetentionPolicy')
-          // We need '*' here because we will also put a retention policy on
-          // the log group of the provider function. Referencing it's name
-          // creates a CF circular dependency.
-          .addAllResources()
-      );
-    }
+    provider.addToRolePolicy(
```

Comment:
add a note here that duplicate statements will be deduplicated by `PolicyDocument`